### ISSUE TITLE
`helm list` shows error

### ISSUE THREAD

**Issue by rimusz** [kubernetes/helm#1173]
`helm list` does not work latest alpha 4 
```
$ helm list
2016/09/09 14:46:13 transport: http2Client.notifyError got notified that the client transport was broken EOF.
Error: transport is closing
```

**Comment by devth** [comment]
Same problem here.

tiller-deploy has:

```yaml
      - image: gcr.io/kubernetes-helm/tiller:v2.0.0-alpha.4
```

and client is:

```
helm version
v2.0.0-alpha.4+36606cf
```

**Comment by fibonacci1729** [comment]
This error can stem from a variety of runtime errors, connection timeouts, etc. Is this happening locally? Steps to reproduce or logs would be a huge help!

**Comment by rimusz** [comment]
I'm using GKE.
Tried it with a few GKE clusters, still the same issue for me.

**Comment by devth** [comment]
My Tiller is running in GKE. It's been running for weeks. I upgraded it to alpha 4 a few days ago. Before that I was running canary. I've deleted release configmaps, resources, and gotten my tiller in a bad state numerous times. 

When I run `helm ls` locally, the remote Tiller's logs show:

```
2016-09-10 13:45:33.504476 I | Listing all releases with filter
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x408c6e]

goroutine 15823 [running]:
panic(0x1473dc0, 0xc42000a040)
        /usr/local/Cellar/go/1.7/libexec/src/runtime/panic.go:500 +0x1a1
main.(*releaseServer).ListReleases.func1(0xc420348a80, 0x14b06a0)
        /Users/adamreese/p/go/src/k8s.io/helm/cmd/tiller/release_server.go:87 +0x1e
k8s.io/helm/pkg/storage.All.func1(0xc420348a80, 0xc4205b66c0)
        /Users/adamreese/p/go/src/k8s.io/helm/pkg/storage/filter.go:51 +0x59
k8s.io/helm/pkg/storage.FilterFunc.Check(0xc4205b66c0, 0xc420348a80, 0x1)
        /Users/adamreese/p/go/src/k8s.io/helm/pkg/storage/filter.go:30 +0x35
k8s.io/helm/pkg/storage.(*Storage).ListFilterAll.func1(0xc420348a80, 0x0)
        /Users/adamreese/p/go/src/k8s.io/helm/pkg/storage/storage.go:94 +0x5a
k8s.io/helm/pkg/storage/driver.(*ConfigMaps).List(0xc42029c2d0, 0xc42056c280, 0x1, 0x13d2f80, 0xc4205aa1e0, 0x0, 0xc420067b00)
        /Users/adamreese/p/go/src/k8s.io/helm/pkg/storage/driver/cfgmaps.go:108 +0x282
k8s.io/helm/pkg/storage.(*Storage).ListFilterAll(0xc42029c300, 0xc42002a020, 0x1, 0x1, 0x0, 0x0, 0xc42027c178, 0x16bf886, 0xc42027c178)
        /Users/adamreese/p/go/src/k8s.io/helm/pkg/storage/storage.go:95 +0x10d
main.(*releaseServer).ListReleases(0xc4201921a0, 0xc4205ca870, 0x1e970a0, 0xc4205aa1c0, 0x0, 0xc42049bcd0)
        /Users/adamreese/p/go/src/k8s.io/helm/cmd/tiller/release_server.go:92 +0xf3
k8s.io/helm/pkg/proto/hapi/services._ReleaseService_ListReleases_Handler(0x1592440, 0xc4201921a0, 0x1e957e0, 0xc42027c100, 0x0, 0xc4204a2000)
        /Users/adamreese/p/go/src/k8s.io/helm/pkg/proto/hapi/services/tiller.pb.go:529 +0x117
k8s.io/helm/vendor/google.golang.org/grpc.(*Server).processStreamingRPC(0xc42033c500, 0x1e971c0, 0xc4205d6000, 0xc4202940e0, 0xc420345680, 0x1eb2720, 0xc42058e840, 0x0, 0x0)
        /Users/adamreese/p/go/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:575 +0x22a
k8s.io/helm/vendor/google.golang.org/grpc.(*Server).handleStream(0xc42033c500, 0x1e971c0, 0xc4205d6000, 0xc4202940e0, 0xc42058e840)
        /Users/adamreese/p/go/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:650 +0xc33
k8s.io/helm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4205aa0a0, 0xc42033c500, 0x1e971c0, 0xc4205d6000, 0xc4202940e0)
        /Users/adamreese/p/go/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:323 +0xab
created by k8s.io/helm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
        /Users/adamreese/p/go/src/k8s.io/helm/vendor/google.golang.org/grpc/server.go:324 +0xa3
```

**Comment by adamreese** [comment]
When helm list gets the configmaps it is not filtering for tiller configmaps.  It is grabbing ALL configmaps in the namespace

The panic happens at https://github.com/kubernetes/helm/blob/master/pkg/storage/driver/cfgmaps.go#L132

verbose logs show an extra configmap for gke
https://gist.github.com/adamreese/e8d336bef6d6cc20c3f16f00a3b21c1b

**Comment by fibonacci1729** [comment]
fixed by #1207
